Userland/Libraries/LibWeb/HTML/DragEvent.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2024, Maciej <sppmacd@pm.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/DragEventPrototype.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/HTML/DragEvent.h>
+
+namespace Web::HTML {
+
+JS_DEFINE_ALLOCATOR(DragEvent);
+
+JS::NonnullGCPtr<DragEvent> DragEvent::create(JS::Realm& realm, FlyString const& event_name, DragEventInit const& event_init)
+{
+    return realm.heap().allocate<DragEvent>(realm, realm, event_name, event_init);
+}
+
+WebIDL::ExceptionOr<JS::NonnullGCPtr<DragEvent>> DragEvent::construct_impl(JS::Realm& realm, FlyString const& event_name, DragEventInit const& event_init)
+{
+    return create(realm, event_name, event_init);
+}
+
+DragEvent::DragEvent(JS::Realm& realm, FlyString const& event_name, DragEventInit const& event_init)
+    // FIXME: What are the correct values for page_x, page_y, offset_x, offset_y?
```

Comment:
For mouse events these values are passed to `MouseEvent::create()`, but are defaulted to 0. Is that something we could do here?

FYI, for `MouseEvent` these values are initially provided by `EventHandler`:

https://github.com/LadybirdBrowser/ladybird/blob/31c7e98a4a46c2d0ef93c5fca47d64d05b96449f/Userland/Libraries/LibWeb/Page/EventHandler.cpp#L272-L275

Reply:
Moved these params up to the constructor / `DragEvent::create`.